tokens/src/core/components/Card.json

```diff
@@ -1,95 +1,82 @@
-        "count": { "value": "3", "type": "dimension", "source": "$card-columns-count" },
-        "gap": { "value": "1.25rem", "type": "dimension", "source": "$card-columns-gap" }
+        "margin": { "source": "$card-columns-margin", "$value": "{spacing.card.spacer.y}" },
+        "count": { "source": "$card-columns-count", "$value": "3" },
+        "gap": { "source": "$card-columns-gap", "$value": "1.25rem" }
       },
       "footer": {
-        "action-gap": { "value": ".5rem", "type": "dimension", "source": "$card-footer-actions-gap" }
+        "action-gap": { "source": "$card-footer-actions-gap", "$value": ".5rem" }
       },
       "loading-skeleton": {
-        "spacer": { "value": ".313rem", "type": "dimension", "source": "$loading-skeleton-spacer" }
+        "spacer": { "source": "$loading-skeleton-spacer", "$value": ".313rem" }
       },
       "logo": {
         "left-offset": {
-          "base": { "value": "1.5rem", "type": "dimension", "source": "$card-logo-left-offset" },
-          "horizontal": { "value": ".4375rem", "type": "dimension", "source": "$card-logo-left-offset-horizontal" }
+          "base": { "source": "$card-logo-left-offset", "$value": "1.5rem" },
+          "horizontal": { "source": "$card-logo-left-offset-horizontal", "$value": ".4375rem" }
         },
         "bottom-offset": {
-          "base": { "value": "1rem", "type": "dimension", "source": "$card-logo-bottom-offset" },
-          "horizontal": { "value": ".4375rem", "type": "dimension", "source": "$card-logo-bottom-offset-horizontal" }
+          "base": { "source": "$card-logo-bottom-offset", "$value": "1rem" },
+          "horizontal": { "source": "$card-logo-bottom-offset-horizontal", "$value": ".4375rem" }
         }
       },
       "focus": {
-        "border-offset": { "value": "5px", "type": "dimension", "source": "$card-focus-border-offset" }
+        "border-offset": { "source": "$card-focus-border-offset", "$value": "5px" }
       }
     }
   },
   "size": {
     "card": {
       "border": {
-        "width": { "value": "{size.border.width}", "type": "dimension", "source": "$card-border-width" },
+        "width": { "source": "$card-border-width", "$value": "{size.border.width}" },
         "radius": {
-          "base": { "value": "{size.border.radius.base}", "type": "dimension", "source": "$card-border-radius" },
-          "logo": { "value": ".25rem", "type": "dimension", "source": "$card-logo-border-radius" },
-          "inner": {
-            "value": "calc({size.card.border.radius.base} - {size.card.border.width})",
-            "type": "dimension",
-            "source": "$card-inner-border-radius"
-          }
+          "base": { "source": "$card-border-radius", "$value": "{size.border.radius.base}" },
+          "logo": { "source": "$card-logo-border-radius", "$value": ".25rem" },
+          "inner": { "source": "$card-inner-border-radius", "$value": "calc({size.card.border.radius.base} - {size.card.border.width})" }
```

Comment:
[curious] Have we looked into whether we can avoid including CSS-specific syntax `calc` in the token itself, or how such tokens would be transformed outside of CSS variables?

For example, if there is arithmetic like subtraction as in this line, could the token `$value` be `{size.card.border.radius.base} - {size.card.border.width}` and have some sort of transform for the CSS variables output to wrap it with `calc(...)`?

That way, tokens like these would be more portable to other non-CSS based platforms like mobile.

Reply:
Dug into this a bit. It appears this still remains a complex problem within the `style-dictionary` community. There is support for a `resolveMath` transform, from `tokens-studio/sd-transforms`; however, when the token outputs references via `outputReferences: true | fn`, the transformed/resolved math is replaced with the original reference (i.e., a CSS variable with `var(--foo: 5px)` in this case).

I attempted to craft a solution for https://github.com/tokens-studio/sd-transforms/issues/203, but came to the same realization recently described [here](https://github.com/amzn/style-dictionary/issues/1055#issuecomment-2312072149):

> The big issue is that outputting refs happens on the format lifecycle and wrapping values in calc() statements usually happens in the transform lifecycle that happens before. So even if you wrap the values in calc() in transform, the format part will just undo that work by outputting refs by using the original value.
>
> Either the calc wrapping needs to happen on the format level or the outputting refs util needs to act on the transformed value somehow (keeping in mind the bugs that this used to cause in v3 and not regressing on this again)
>
> There are a couple of open issues to rethink how references are resolved and how values with references in them are transformed for a future v5 version of Style Dictionary, but this topic is rather complex and this issue won't be fixed until we have solutions to that broader topic.

tl;dr; Handling the wrapping of math expressions with the CSS `calc` syntax remains an open question, and likely will not land in `v4` of Style Dictionary per the above. Given this, it probably makes sense to keep the `calc` syntax for now, but when we begin transforming to non-CSS platforms (e.g., JavaScript, iOS, Android), having `calc` in the token value itself will present an issue. A possible workaround for if/when this becomes an issue might be to introduce custom transforms for the (future) non-CSS platforms that strips the wrapping `calc(...)` from the underlying math operations.

Reply:
This is definitely a very interesting question that we haven't thought about yet. Thanks for your research, now we know more!